code/modules/reagents/chemistry/items.dm

```diff
@@ -0,0 +1,162 @@
+///if the pHmeter gives a detailed output
+#define DETAILED_CHEM_OUTPUT 1 
+///if the pH meter gives a shorter output
+#define SHORTENED_CHEM_OUTPUT 0 
+
+/*
+* a pH booklet that contains pH paper pages that will change color depending on the pH of the reagents datum it's attacked onto
+*/
+/obj/item/pHbooklet
+	name = "pH indicator booklet"
+	desc = "A booklet containing paper soaked in universal indicator."
+	icon_state = "pHbooklet"
+	icon = 'icons/obj/chemical.dmi'
+	item_flags = NOBLUDGEON
+	resistance_flags = FLAMMABLE
+	w_class = WEIGHT_CLASS_TINY
+	///How many pages the booklet holds
+	var/number_of_pages = 50
+
+//A little janky with pockets
+/obj/item/pHbooklet/attack_hand(mob/user, act_intent = user.a_intent, unarmed_attack_flags)
+	if(user.get_held_index_of_item(src))//Does this check pockets too..?
+		if(number_of_pages == 50)
+			icon_state = "pHbooklet_open"
+		if(number_of_pages >= 1)
+			var/obj/item/pHpaper/P = new /obj/item/pHpaper
+			P.add_fingerprint(user)
+			P.forceMove(user.loc)
+			user.put_in_active_hand(P)
+			to_chat(user, "<span class='notice'>You take [P] out of \the [src].</span>")
+			number_of_pages--
+			playsound(user.loc, 'sound/items/poster_ripped.ogg', 50, TRUE)
+			add_fingerprint(user)
+			if(number_of_pages == 0)
+				icon_state = "pHbooklet_empty"
+			return
+		else
+			to_chat(user, "<span class='warning'>[src] is empty!</span>")
+			add_fingerprint(user)
+			return
+	. = ..()
+	var/I = user.get_active_held_item()
+	if(!I)
+		user.put_in_active_hand(src)
+
+/obj/item/pHbooklet/MouseDrop()
+	var/mob/living/user = usr
+	if(number_of_pages >= 1)
+		var/obj/item/pHpaper/P = new /obj/item/pHpaper
+		P.add_fingerprint(user)
+		P.forceMove(user)
+		user.put_in_active_hand(P)
+		to_chat(user, "<span class='notice'>You take [P] out of \the [src].</span>")
+		number_of_pages--
+		playsound(user.loc, 'sound/items/poster_ripped.ogg', 50, TRUE)
+		add_fingerprint(user)
+		if(number_of_pages == 0)
+			icon_state = "pHbookletEmpty"
+		return
+	else
+		to_chat(user, "<span class='warning'>[src] is empty!</span>")
+		add_fingerprint(user)
+		return
+
+/*
+* pH paper will change color depending on the pH of the reagents datum it's attacked onto
+*/
+/obj/item/pHpaper
+	name = "pH indicator strip"
+	desc = "A piece of paper that will change colour depending on the pH of a solution."
+	icon_state = "pHpaper"
+	icon = 'icons/obj/chemical.dmi'
+	item_flags = NOBLUDGEON
+	color = "#f5c352"	
+	resistance_flags = FLAMMABLE
+	w_class = WEIGHT_CLASS_TINY
+	///If the paper was used, and therefore cannot change color again
+	var/used = FALSE
+
+/obj/item/pHpaper/afterattack(obj/item/reagent_containers/cont, mob/user, proximity)
+	if(!istype(cont))
+		return
```

Comment:
```suggestion
/obj/item/pHpaper/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
	if(!istype(cont, /obj/item/reagent_containers))
		return
	var/obj/item/reagent_containers/cont = target
		return
```
* Keep all the arguments defined on proc redefinitions, even if you don't use them.
* Don't change the incoming type, so future coders get a clear picture of what kind of input they can expect from this proc calls.

Reply:
Thanks for pointing this out